lib/oa/graph.rb

```diff
@@ -78,30 +78,17 @@ def id_as_url
     # @return [Array<String>] Array of urls expressing the OA motivated_by
     #   values
     def motivated_by
-      motivations = []
       q = self.class.anno_query.dup
       q << [:s, RDF::Vocab::OA.motivatedBy, :motivated_by]
-      solution = @graph.query q
-      if solution && solution.size > 0
-        solution.each { |res|
-          motivations << res.motivated_by.to_s
-        }
-      end
-      # TODO:  raise exception if none? (validation)
-      motivations
+      # if the motivation object gives junk with .to_s, then the anno is illegal anyway and we're not responsible
+      @graph.query(q).each.collect {|stmt| stmt.motivated_by.to_s}
```

Comment:
Yes.  Perhaps we need a new issue to focus on whether to coerce the result with `.to_s`.